Pairing issue with the latest SDK commit using all-clusters-app

Issue :- Not able to advertise the network (ble-wifi/onnetwork) from dut to get paired with controller using all-clusters-app.


Commit used - a75cd99d851cff80e14e753c85f34ff936c84254

commands used for advertising (bringing up dut) : 

Ble-wifi :` connectedhomeip/examples/all-clusters-app/linux/out/all-clusters-app$ ./chip-all-clusters-app --wifi`

Onnetwork : `connectedhomeip/examples/all-clusters-app/linux/out/all-clusters-app$ ./chip-all-clusters-app `

Log:

> ubuntu@matter-te-7:~/connectedhomeip/examples/all-clusters-app/linux/out/all-clusters-app$ ./chip-all-clusters-app 
INTERNAL ERROR: No help text defined for option: -
[1649856140.280266][8986:8986] CHIP:-: Failed to init Linux App: ../../third_party/connectedhomeip/examples/platform/linux/Options.cpp:409: Error 0x0000002F 
[1649856140.280328][8986:8986] CHIP:SPT: VerifyOrDie failure at ../../main.cpp:25: ChipLinuxAppInit(argc, argv, AppOptions::GetOptions()) == 0
Aborted (core dumped)

> ubuntu@matter-te-7:~/connectedhomeip/examples/all-clusters-app/linux/out/all-clusters-app$ ./chip-all-clusters-app --wifi
INTERNAL ERROR: No help text defined for option: -
[1649856349.791896][8990:8990] CHIP:-: Failed to init Linux App: ../../third_party/connectedhomeip/examples/platform/linux/Options.cpp:409: Error 0x0000002F 
[1649856349.791954][8990:8990] CHIP:SPT: VerifyOrDie failure at ../../main.cpp:25: ChipLinuxAppInit(argc, argv, AppOptions::GetOptions()) == 0
Aborted (core dumped)

--- a/src/lib/support/CHIPArgParser.cpp
+++ b/src/lib/support/CHIPArgParser.cpp
@@ -77,8 +77,6 @@ static void PutStringWithNewLine(FILE * s, const char * str);
 static void PutStringWithBlankLine(FILE * s, const char * str);
 #if CHIP_CONFIG_ENABLE_ARG_PARSER_VALIDITY_CHECKS
 static bool SanityCheckOptions(OptionSet * optSets[]);
-static bool HelpTextContainsLongOption(const char * optName, const char * helpText);
-static bool HelpTextContainsShortOption(char optChar, const char * helpText);
 #endif // CHIP_CONFIG_ENABLE_ARG_PARSER_VALIDITY_CHECKS
 
 static inline bool IsShortOptionChar(int ch)
@@ -1511,51 +1509,7 @@ static bool SanityCheckOptions(OptionSet * optSets[])
                     }
             }
 
-    // Fail if the option help texts do not contain a description for each option, including both
-    // the option's long and short forms.
-    for (OptionSet ** optSetP = optSets; *optSetP != nullptr; optSetP++)
-        for (OptionDef * optionDef = (*optSetP)->OptionDefs; optionDef->Name != nullptr; optionDef++)
-        {
-            if (!HelpTextContainsLongOption(optionDef->Name, (*optSetP)->OptionHelp))
-            {
-                PrintArgError("INTERNAL ERROR: No help text defined for option: --%s\n", optionDef->Name);
-                res = false;
-            }
-
-            if (IsShortOptionChar(optionDef->Id) &&
-                !HelpTextContainsShortOption(static_cast<char>(optionDef->Id), (*optSetP)->OptionHelp))
-            {
-                PrintArgError("INTERNAL ERROR: No help text defined for option: -%c\n", optionDef->Id);
-                res = false;
-            }
-        }
-
     return res;
-}
-
-static bool HelpTextContainsLongOption(const char * optName, const char * helpText)
-{
-    size_t nameLen = strlen(optName);
-
-    for (const char * p = helpText; (p = strstr(p, optName)) != nullptr; p += nameLen)
-        if ((p - helpText) >= 2 && p[-1] == '-' && p[-2] == '-' && !isalnum(p[nameLen]) && p[nameLen] != '-')
-            return true;
-
-    return false;
-}
-
-static bool HelpTextContainsShortOption(char optChar, const char * helpText)
-{
-    char optStr[3];
-    optStr[0] = '-';
-    optStr[1] = optChar;
-    optStr[2] = 0;
-
-    for (const char * p = helpText; (p = strstr(p, optStr)) != nullptr; p += 2)
-        if ((p == helpText || (!isalnum(p[-1]) && p[-1] != '-')) && !isalnum(p[2]) && p[2] != '-')
-            return true;
-
-    return false;
 }
 
 #endif // CHIP_CONFIG_ENABLE_ARG_PARSER_VALIDITY_CHECKS

